### ISSUE TITLE
[14.0] signup/password reset spam not being blocked by recaptcha

### ISSUE THREAD

**Issue by th3penguinwhisperer** [odoo/odoo#185423]
### Odoo Version

- [X] 16.0
- [X] 17.0
- [X] 18.0
- [X] Other (specify)

### Steps to Reproduce

I see frequent signup spam on Odoo. Users keep on getting created from IPs all over the world. This triggers emails to be sent out to confirm the email address and the website user list have plenty of these accounts.

I can't delete these users, nor archive them as they have a partner link (why have a button to delete a user if it's impossible?). Even when I remove the portal access it's still impossible to remove/archive those users.

Obviously this should be resolved at the core by having a basic Recaptcha but for some reason this was only implemented for forms and not for signup and password reset.

How do I resolve this? (no paid addons for basic functionality that's already basically implemented for forms)
How can I properly remove these users? Preferably without needing to go manually deleting in the database and possibly introducing issues down the line.

Tried on the forum but my post doesn't get validated for a week already.

**Comment by dansanti** [comment]
I agree , reCaptcha must be integrated in signup/reset password,  i been  already blocked from gmail by spammers that make brute force to that options.

And i have a lot of fake accounts in brute force attempts

**Comment by th3penguinwhisperer** [comment]
@bram1000 I was wondering what the difference is between 185648 and 185073?
They both have the same idea but one is for 18 and the other one is for older versions?

**Comment by th3penguinwhisperer** [comment]
Hi @bram1000 
Is there anything merged on any branches? 14.0 15.0 16.0 17.0 18.0?
We're months down the line and at this point google is actively blocking domains for spamming due to Odoo...
There seems to be no end in sight on something that should have been part of the base a decade ago.

**Comment by mof-landoo** [comment]
Hello @bram1000 I think this is already solved in 17.0 and 18.0, thanks a lot.
Do you know when will be available in 16.0?

**Comment by kamzata** [comment]
Any news on this?

**Comment by ctodavid** [comment]
I have it in version 17 and I suffer it continuously so I remove the free registration and only leave it by invitation, in the end this means that I can't sell through the marketplace, I don't understand how they don't solve it.
